computervision/src/main/java/ComputerVision.kt

```diff
@@ -25,11 +28,11 @@ sealed class Vision : SensorReading {
     class Scenery(val scene: String) : Vision()
     class Facial (val prettyFace: String) : Vision()
     class Emotion (val emotion: String) : Vision()
-    class ObjectDetection (val objects: List<Any>): Vision()
+    class ObjectDetection (val objects: JSONArray): Vision()
 }
 
 sealed class CVInput : SensorInput {
-    class ModeChange(val mode: Int) : CVInput()
+    class ModeChange(val mode: JSONObject) : CVInput()
 
 
```

Comment:
Big No! don't ever expose the way you parse/tokenize data in your public interfaces/objects; this is an internal detail, and users of your API need not know about it. Plus, this way, if you need to change the way you parse you''ll have to change the code in here too, which means users of your API will need to change their code, too!
Leave the tokenizing business to the `tokenize`r.

Reply:
Okay!